classes/components/forms/submission/ChangeSubmissionLanguageMetadataForm.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/**
+ * @file classes/components/form/submission/ChangeSubmissionLanguageMetadataForm.php
+ *
+ * Copyright (c) 2024 Simon Fraser University
+ * Copyright (c) 2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class ChangeSubmissionLanguageMetadataForm
+ *
+ * @brief A form for changing submission's metadata before language change.
+ */
+
+namespace PKP\components\forms\submission;
+
+use APP\facades\Repo;
+use APP\publication\Publication;
+use APP\submission\Submission;
+use PKP\components\forms\Field;
+use PKP\components\forms\FieldHTML;
+use PKP\components\forms\FieldRadioInput;
+use PKP\components\forms\FormComponent;
+use PKP\components\forms\publication\TitleAbstractForm;
+use PKP\context\Context;
+use PKP\facades\Locale;
+
+class ChangeSubmissionLanguageMetadataForm extends FormComponent
+{
+    public const FORM_CHANGE_SUBMISSION_LANGUAGE_METADATA = 'changeSubmissionLanguageMetadata';
+    public $id = self::FORM_CHANGE_SUBMISSION_LANGUAGE_METADATA;
+    public $method = 'PUT';
+
+    public function __construct(string $action, Submission $submission, Publication $publication, Context $context)
+    {
+        $this->action = $action;
+
+        $this->addGroup([
+            'id' => 'language',
+        ]);
+
+        $submissionLocale = $submission->getData('locale');
+        $supportedLocaleNames = collect($context->getSupportedSubmissionLocaleNames());
+        $submissionLocaleNameArr = collect(Locale::getSubmissionLocaleDisplayNames([$submissionLocale]));
```

Comment:
I know it's small, but having this appended with `Arr` at the end makes it seem like it's different from the `$supportedLocaleNames` Eloquent collection. Especially with all the transformations going on, I would suggest keeping the naming conventions the same and calling this $submissionLocaleNames` or something like that.